openwisp_utils/admin_theme/static/admin/js/ow-filter.js

```diff
@@ -0,0 +1,185 @@
+'use strict';
+var leftArrow, rightArrow, slider;
+const scrollDX = 200,
+  btnAnimationTime = 100; //ms
+(function () {
+  document.addEventListener(
+    'DOMContentLoaded',
+    function () {
+      leftArrow = document.querySelector('.filters-bottom .left-arrow');
+      rightArrow = document.querySelector('.filters-bottom .right-arrow');
+      slider = document.querySelector('.ow-filter-slider');
+      initFilterDropdownHandler();
+      initSliderHandlers();
+      filterHandlers();
+      setArrowButtonVisibility();
+    },
+    false
+  );
+})();
+
+function initFilterDropdownHandler() {
+  const filters = document.querySelectorAll('.ow-filter');
+  // When filter title is clicked
+  filters.forEach(function (filter) {
+    var toggler = filter.querySelector('.filter-title');
+    toggler.addEventListener('click', function () {
+      // Close if any active filter
+      var activeFilter = document.querySelector('.ow-filter.active');
+      if (activeFilter && activeFilter !== filter) {
+        activeFilter.classList.remove('active');
+      }
+      filter.classList.toggle('active');
+    });
+  });
+  // Handle click outside of an active filter
+  document.addEventListener('click', function (e) {
+    var activeFilter = document.querySelector('.ow-filter.active');
+    if (activeFilter && !activeFilter.contains(e.target)) {
+      activeFilter.classList.remove('active');
+    }
+  });
+  // Handle change in filter option
+  const filterRadios = document.querySelectorAll('.filter-options input');
+  filterRadios.forEach(function (filterRadio) {
+    filterRadio.addEventListener('change', function () {
+      let filter = document.querySelector('.ow-filter.active');
+      let view = document.querySelector('.ow-filter.active .selected-option');
+      let selectedElement = document.querySelector(
+        '.ow-filter.active .selected'
+      );
+      selectedElement.classList.remove('selected');
+      filterRadio.previousElementSibling.classList.add('selected');
+      var text = filterRadio.previousElementSibling.innerHTML;
+      view.innerHTML = text;
+      filter.querySelector('.filter-title').setAttribute('title', text);
+      filter.classList.remove('active');
+    });
+  });
+}
+
+function ButtonAnimation(button) {
```

Comment:
why are you using uppercase first letter for a function?  Avoid this please, the convention is to use this for classes.